WordPress/src/main/java/org/wordpress/android/ui/stories/StoriesMediaPickerResultHandler.kt

```diff
@@ -26,7 +25,7 @@ class StoriesMediaPickerResultHandler {
                         PagePostCreationSourcesDetail.STORY_FROM_MY_SITE
                 )
                 return true
-            } else if (isWPStoriesMediaBrowserTypeResult(data)) {
```

Comment:
I think we actually still need to discriminate if we are coming here from a stories picker or not, otherwise we risk to trigger the stories path even when not needed. For example:

- In My Site change the site icon
- In the picker select an image for the site icon and Confirm
- In this case we are landed to the story path but we shouldn't

Reply:
this is a great catch 👍 . I've solved this using 2 approaches for when the flag is on/off:
- When the flag is off, the behaviour should be exactly as it was before
- When the flag is on, there is a new RequestCode called STORIES_PHOTO_PICKER which is triggered only when we're creating a story. This was we stop mixing up the normal photo picker and the stories photo picker and the need for the check to `isWPStoriesMediaBrowserTypeResult` disappears. 

To test it check the flow both with and without the flag.